Makefile

```diff
@@ -191,7 +191,7 @@ delete-tag:
 docker-otelcol:
 	GOOS=linux $(MAKE) otelcol
 	cp ./bin/otelcol_linux_amd64 ./cmd/otelcol/otelcol
-	docker build -t otelcol ./cmd/otelcol/
+	docker build -t otelcol ./cmd/otelcol/ --build-arg SMART_AGENT_RELEASE_URL=https://github.com/signalfx/signalfx-agent/releases/download/$(SMART_AGENT_RELEASE)/signalfx-agent-$(subst v,,$(SMART_AGENT_RELEASE)).tar.gz
```

Comment:
Instead of passing the url as a build arg, would it be simpler to pass `SMART_AGENT_RELEASE` as a build arg, and set the url in the dockerfile, i.e. `--build-arg SMART_AGENT_RELEASE=$(SMART_AGENT_RELEASE)`?

Also, we'll need to set a default for `SMART_AGENT_RELEASE` either in the Makefile, and/or update the dockerfile to handle `SMART_AGENT_RELEASE` correctly if it is empty, so that we can run `make docker-otelcol` without any args.

Reply:
> Also, we'll need to set a default for `SMART_AGENT_RELEASE` either in the Makefile, and/or update the dockerfile to handle `SMART_AGENT_RELEASE` correctly if it is empty, so that we can run `make docker-otelcol` without any args.

So, what should the default SMART_AGENT_RELEASE be? v5.9.1?

Reply:
For now, I think v5.9.1 is fine since that is what the deb, rpm, and msi packages include.  I'll defer to @rmfitzpatrick if we want to upgrade to the latest v5.9.2 release.